app/controllers/notification/ReceivedALetterController.scala

```diff
@@ -62,11 +63,27 @@ class ReceivedALetterController @Inject()(
         formWithErrors =>
           Future.successful(BadRequest(view(formWithErrors, mode))),
 
-        value =>
+        value => { 
+          
+          val (pagesToClear, hasValueChanged) = changedPages(request.userAnswers, value)
+
           for {
             updatedAnswers <- Future.fromTry(request.userAnswers.set(ReceivedALetterPage, value))
-            _              <- sessionRepository.set(updatedAnswers)
-          } yield Redirect(navigator.nextPage(ReceivedALetterPage, mode, updatedAnswers))
+            clearedAnswers <- Future.fromTry(updatedAnswers.remove(pagesToClear))
+            _              <- sessionRepository.set(clearedAnswers)
+          } yield Redirect(navigator.nextPage(ReceivedALetterPage, mode, clearedAnswers, hasValueChanged))
+        }
       )
   }
+
+  def changedPages(userAnswers: UserAnswers, value: Boolean): (List[QuestionPage[_]], Boolean) = {
+    userAnswers.get(ReceivedALetterPage) match {
+      case Some(false) if value == true =>
+        (Nil, true)
+      case Some(true) if value == false =>
+        (removeLetterReferencePages, false)
```

Comment:
As this isn't likely to be reused for other pages I think we can just say ```List(LetterReferencePage)``` here instead of referencing a val in the trait

Reply:
resolved!!